### DISCUSSION TITLE
PyTorch and Detect.py get different results?

### DISCUSSION THREAD

**Discussion by alevangel** [ultralytics/yolov5#4617] [Q&A] [answered]
Hello all, 
I'm very new to the matter, I'm trying to use the YOLOv5 with PyCharm. In roder to do that i'm using the PyThorch Hub, but when I do a simple model(img) I get very different results from detect.py executed with a notebook. 

I also configured the correct size of image (that in my case is 1028) and custom weights, this a snippet:
```
# Model
model = torch.hub.load(PATH+'yolov5', 'custom', path=PATH+'best.pt'
                       , force_reload=True
                       , source='local')
# Image
img = PATH+'test/images/66.jpg'

# Inference
model.conf = 0.55
results = model(img, size=1280, augment=True)
```

While this is what I do with detect.py:
`!python detect.py --weights runs/train/yolov5m_results14/weights/best.pt --img 1024 --conf 0.55 --source /content/test/images`

Actually I get different confidence levels and also different boxes, why this happens?

**Comment by glenn-jocher** [comment, marked as answer]
@alevangel you're running inference with different image sizes.

**Reply by alevangel** [reply]
Thank you so much... I don't why was struggling so much about it! D: In my biased mind were equal numbers!

**Reply by glenn-jocher** [reply]
@alevangel also augmented inference is very different, it's going to produce better results but take longer. The augment=True addition to your hub inference is doing this while detect.py is not augmented in your command.

**Reply by alevangel** [reply]
@glenn-jocher Uh I see, thanks! I thought it was augmented by default with detect.py, how to set it?

**Reply by glenn-jocher** [reply]
@alevangel see detect.py argarser for inference options:
```
python detect.py --augment
```

https://github.com/ultralytics/yolov5/blob/de534e922120b2da876e8214b976af1f82019e28/detect.py#L248-L277
